src/postamble.js

```diff
@@ -29,6 +32,8 @@ function callMain() {
 #if ASSERTIONS
   assert(runDependencies == 0, 'cannot call main when async dependencies remain! (listen on Module["onRuntimeInitialized"])');
   assert(__ATPRERUN__.length == 0, 'cannot call main when preRun functions remain to be called');
+  assert(!alreadyCalledMain, 'cannot call main more than once (consider using a library, or MODULARIZE)');
+  alreadyCalledMain = true;
```

Comment:
I think some folks might be depending on this.    Should we use `warnOnce` instead?   Or perhaps if folks really want to do this they can set `alreadyCalledMain = false` each time the call it?